src/modelmanager.hpp

```diff
@@ -367,6 +405,35 @@ class ModelManager {
      * @brief Updates OVMS configuration with cached configuration file. Will check for newly added model versions
      */
     Status updateConfigurationWithoutConfigFile();
+
+    /**
+     * @brief Cleaner thread procedure to cleanup resources that are not used
+     */
+    void cleanupResources();
 };
 
+struct FunctorSequenceCleaner {
+    GlobalSequencesViewer& globalSequencesViewer;
+
+    FunctorSequenceCleaner(GlobalSequencesViewer& globalSequencesViewer) :
+        globalSequencesViewer(globalSequencesViewer) {}
+
+    virtual void cleanup() {
+        globalSequencesViewer.removeIdleSequences();
+    }
+};
+
+struct FunctorResourcesCleaner {
+    ModelManager& modelManager;
+
+    FunctorResourcesCleaner(ModelManager& modelManager) :
+        modelManager(modelManager) {}
+
+    virtual void cleanup() {
+        modelManager.cleanupResources();
+    }
+};
```

Comment:
This should be in separate header. Maybe cleaner_utils.hpp? This way other components using MM does not know about way how MM handles cleanup.